### PR TITLE
Extend date object of `sugar.js` library

### PR DESCRIPTION
**Pull request by HNOONa-0**
# Description

I installed `sugar` library and "extended" the `Date` object so that it could have the same effect as `sugar.js` script.
I also added a `Jest` mock so that `Jest` could recognize this package, its method and function calls. Changes are working as expected.

Fixes: #7753 

# Comments

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Review by pabo99 — COMMENTED** [review]
Excellent!

One last comment.

We still have an issue to resolve:

In the file `head.sugar_locale.js`, the `locale` value is globally set to `en` when the `pseudo` language is chosen in user preferences because it is not a valid locale:


https://github.com/omegaup/omegaup/blob/b9451e18e132ca363a4315b48cd9a59fa65c8a2b/frontend/www/js/head.sugar_locale.js#L1-L7

This should not affect much anymore since we no longer display that option in the dropdown. However, we can still access this language via the following URL for instance:

http://localhost:8001/arena/?lang=pseudo

With the changes you made, that global variable is no longer stored. Therefore, when trying to set the `locale` value from the `time.ts` library, it attempts to set it again as `pseudo`.

I can think of two possible solutions (both allowing us to remove the `head.sugar_locale.js` file):

- Move the logic from this file to the `setSugarLocale` function, where you can access the `data-locale` attribute and change the `locale` value there.

- Similar to what you did for the Spanish and Portuguese languages, you can add a `locales/pseudo.ts` file containing the same information in English

**Comment by HNOONa-0** [comment]
> Excellent!
> 
> One last comment.
> 
> We still have an issue to resolve:
> 
> In the file `head.sugar_locale.js`, the `locale` value is globally set to `en` when the `pseudo` language is chosen in user preferences because it is not a valid locale:
> 
> https://github.com/omegaup/omegaup/blob/b9451e18e132ca363a4315b48cd9a59fa65c8a2b/frontend/www/js/head.sugar_locale.js#L1-L7
> 
> This should not affect much anymore since we no longer display that option in the dropdown. However, we can still access this language via the following URL for instance:
> 
> http://localhost:8001/arena/?lang=pseudo
> 
> With the changes you made, that global variable is no longer stored. Therefore, when trying to set the `locale` value from the `time.ts` library, it attempts to set it again as `pseudo`.
> 
> I can think of two possible solutions (both allowing us to remove the `head.sugar_locale.js` file):
> 
> * Move the logic from this file to the `setSugarLocale` function, where you can access the `data-locale` attribute and change the `locale` value there.
> * Similar to what you did for the Spanish and Portuguese languages, you can add a `locales/pseudo.ts` file containing the same information in English

`sugar` ships with `en` locale by default, we can retrieve and apply that config to `pseudo` per documentation.

File: frontend/templates/template.tpl [hunk 1]
```diff
@@ -17,7 +17,6 @@
     <script type="text/javascript" src="{% versionHash '/third_party/js/jquery-3.5.1.min.js' %}"></script>
     <script type="text/javascript" src="{% versionHash '/js/jquery_error_handler.js' %}"></script>
     <script type="text/javascript" src="{% versionHash '/third_party/js/highstock.js' %}" defer></script>
-    <script type="text/javascript" src="{% versionHash '/third_party/js/sugar.js' %}"></script>
```
Comment (pabo99) [inline frontend/templates/template.tpl:20]: Is it possible to remove this file now? I think it's no longer needed after this change

**Review by pabo99 — COMMENTED** [review]
The very last comment, I promise

**Review by pabo99 — APPROVED** [review]
Thank you!

**Pull request merged**